Description: Workaround WLAN detection
 Allow returning of WLAN information even when network signal strength 
 is not available. It's a workaround to the fact that the kernel driver 
 for mako (Nexus 4) does not return the signal strength ever.
 .
Author: Ricardo Salveti <ricardo.salveti@canonical.com>
Origin: Canonical
Forwarded: https://codereview.qt-project.org/66904
Bug: https://bugreports.qt-project.org/browse/QTBUG-33748
Bug-Ubuntu: https://launchpad.net/bugs/1231196
Last-Update: 2013-09-30

diff --git a/src/systeminfo/linux/qnetworkinfo_linux.cpp b/src/systeminfo/linux/qnetworkinfo_linux.cpp
index effdc93..03649c1 100644
--- a/src/systeminfo/linux/qnetworkinfo_linux.cpp
+++ b/src/systeminfo/linux/qnetworkinfo_linux.cpp
@@ -869,8 +869,7 @@ QNetworkInfo::NetworkStatus QNetworkInfoPrivate::getNetworkStatus(QNetworkInfo::
             if (carrier.open(QIODevice::ReadOnly)) {
                 char state;
                 if ((carrier.read(&state, 1) == 1) &&
-                        (state == '1') &&
-                        (networkSignalStrength(QNetworkInfo::WlanMode, interface) > -1)) {
+                        (state == '1')) {
                     return QNetworkInfo::HomeNetwork;
                 }
             }
